vm/starnix/starnix.go

```diff
@@ -326,26 +326,66 @@ func (inst *instance) Copy(hostSrc string) (string, error) {
 		if inst.debug {
 			log.Logf(1, "instance %s: attempting to push executor binary over ADB", inst.name)
 		}
-		err := inst.runCommand(
+		output, err := osutil.RunCmd(
+			1*time.Minute,
+			inst.fuchsiaDirectory,
 			"adb",
 			"-s",
 			fmt.Sprintf("127.0.0.1:%d", inst.port),
 			"push",
 			hostSrc,
 			vmDst)
 		if err == nil {
-			return vmDst, nil
+			err = inst.Chmod(vmDst)
+			return vmDst, err
 		}
 		if inst.debug {
 			log.Logf(1, "instance %s: adb push failed", inst.name)
 		}
+		// Retryable connection errors usually look like "adb: error: ... : device offline"
+		// or "adb: error: ... closed"
+		if !strings.HasPrefix(string(output), "adb: error:") {
+			return vmDst, err
```

Comment:
Might be helpful to log `output` here for debugging when this case is hit? (Or even log it regardless of prefix if `debug` is enabled, similar to how `inst.runCommand` did previously.)

Reply:
Good idea! Done.